gateway/api_loader.go

```diff
@@ -763,6 +766,9 @@ func loadApps(specs []*APISpec) {
 	shouldTrace := trace.IsEnabled()
 	for _, spec := range specs {
 		func() {
+			if strings.Contains(spec.Proxy.TargetURL, "h2c://") {
+				spec.Protocol = "h2c"
```

Comment:
should we check if `spec.Protocol` is already set? If for instance `spec.Protocol=http` and target is `h2c` is that allowed? If not then spec is not valid we should emit an error or something.

Reply:
@gernest  if http is set, and the target contains "h2c://" then it has major precedence, hence we set h2c as protocol. In the very beggining we did this in this way due that FE doesn't have a protocol option `h2c`.